ocean_lib/data_provider/data_service_provider.py

```diff
@@ -205,33 +187,47 @@ def download_service(
             indexes = [index]
 
         req = PreparedRequest()
-        params = {
-            "documentId": did,
+        service_endpoint += f"?documentId={did}"
```

Comment:
why do we have some parameters added like this, and some by using payload?

Reply:
prepare_url function transforms ':' from "did:op:" into "%3".